### PR TITLE
Package plebeia.2.0.0

### PR DESCRIPTION
**Pull request by camlspotter**
### `plebeia.2.0.0`
Functional storage using Merkle Patricia tree

---
* Homepage: https://gitlab.com/dailambda/plebeia/
* Source repo: git+https://gitlab.com/dailambda/plebeia/
* Bug tracker: https://gitlab.com/dailambda/plebeia/-/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

**Comment by camlspotter** [comment]
> macos-homebrew (experimental)
> [plebeia.2.0.0 (failed: linking with `cc` failed: exit status: 1)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/0b19f7dcff0b9ad3ce7bb5463a5064920aeccd87/variant/distributions,macos-homebrew,plebeia.2.0.0)

I cannot reproduce it in my homebrew...

> s390x
> [plebeia.2.0.0 (failed: `ptr::const_ptr::<impl *const T>::offset` is not yet stable as a const fn)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/0b19f7dcff0b9ad3ce7bb5463a5064920aeccd87/variant/extras,s390x,plebeia.2.0.0)

The `opam` file excludes `s390x` but the CI still tries to compile under it.

File: packages/plebeia/plebeia.2.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "dune" {>= "3.2.0"}
+  "bigstring" {= "0.3"}
+  "hacl-star" {= "0.4.5"}
+  "blake3" {= "0.2"}
+  "cstruct" {= "6.1.0"}
+  "stdint" {= "0.7.0"}
+  "hex" {= "1.5.0"}
+  "data-encoding" {= "0.5.3"}
+  "ptime" {= "1.0.0"}
+  "alcotest" {= "1.5.0"}
```
Comment (kit-ty-kate) [inline packages/plebeia/plebeia.2.0.0/opam:18]: Your constraints seem to be extremely strict and will cause a lot of problems for your users.
```suggestion
  "bigstring" {>= "0.3"}
  "hacl-star" {>= "0.4.5"}
  "blake3" {>= "0.2"}
  "cstruct" {>= "6.1.0"}
  "stdint" {>= "0.7.0"}
  "hex" {>= "1.5.0"}
  "data-encoding" {>= "0.5.3"}
  "ptime" {>= "1.0.0"}
  "alcotest" {>= "1.5.0" & with-test}
```
Also some of your tests dependencies don’t seem to be marked as is

**Comment by kit-ty-kate** [comment]
> The `opam` file excludes `s390x` but the CI still tries to compile under it.

It was puzzling to me but it seems to be a quirk of our CI which uses `opam install --deps-only <pkg>` which ignores the `available` field for `<pkg>` and thus it was failing in `blake3`. I fixed it in https://github.com/ocaml/opam-repository/pull/21895

The macOS issue seems to be a CI setup issue, we can ignore it for now.

**Comment by kit-ty-kate** [comment]
Sorry for the wait. Thanks!

**Pull request merged**